src/example-types/metricGroups/pivot.js

```diff
@@ -31,6 +31,49 @@ let aggsForValues = (node, schema) =>
     }))
   )(node)
 
+let maybeWrapWithFilterAgg = ({ query, filters, aggName }) =>
+  _.isEmpty(filters)
+    ? query
+    : {
+        aggs: {
+          [aggName]: {
+            filter: { bool: { must: filters } },
+            ...query,
+          },
+        },
+      }
+
+let getSortAgg = async ({ node, sort, schema, getStats }) => {
+  if (!sort) return
+
+  let filters = _.compact(
+    await Promise.all(
+      F.mapIndexed((column, i) => {
+        let filter = lookupTypeProp(_.stubFalse, 'drilldown', column.type)
+        let drilldown = sort?.columnValues[i]
+        return drilldown && filter({ drilldown, ...column }, schema, getStats)
+      }, node.columns)
+    )
+  )
```

Comment:
we should unify this with the same code from line ~124 in this file:


```js
let drilldownFilters = async (drilldowns, groups) =>
  _.compact(
    await Promise.all(
      F.mapIndexed((group, i) => {
        let filter = lookupTypeProp(_.stubFalse, 'drilldown', group.type)
        // stamp on drilldown from root if applicable
        let drilldown = drilldowns[i] || group.drilldown
        return drilldown && filter({ drilldown, ...group }, schema, getStats)
      }, groups)
    )
  )
```
and then use it like this:

```js
let thisUseCase = drilldownFilters(sort?.columnValues, node.columns)
let useCaseBelow = drilldownFilters(drilldowns, node.groups)
```

This work needs to be done to support the `filter` use case I've got locally. I can just do it as part of that PR or party on this one.

Reply:
here's some further cleanup I did locally so I could finish off the `filter` use case:

```js
let compactMapIndexedAsync = async (...args) =>
  _.compact(await Promise.all(F.mapIndexed(...args)))

let drilldownFilters = async (drilldowns, groups) =>
  compactMapIndexedAsync((group, i) => {
    let filter = lookupTypeProp(_.stubFalse, 'drilldown', group.type)
    // stamp on drilldown from root if applicable
    let drilldown = drilldowns[i] || group.drilldown
    return drilldown && filter({ drilldown, ...group }, schema, getStats)
  }, groups)

// This PR's use cases:
drilldownFilters(drilldowns, node.groups)
drilldownFilters(sort?.columnValues, node.columns)

// New Functionality:
// node.filters = [
//   { rows: ['Reno', '0-500'], columns: ['2017'] },
//   { rows: ['Hillsboro Beach', '2500-*'] }
// ] -> (Reno AND 0-500) OR (Hillsboro AND 2500-*)
let hasValue = ({ filters }) => !_.isEmpty(filters)
let filter = async ({ filters, groups, columns }) => ({
  bool: {
    minimum_should_match: 1,
    should: await compactMapIndexedAsync(
      async (filter) => ({
        bool: {
          must: [
            ...drilldownFilters(filter.rows, groups),
            ...drilldownFilters(filter.columns, columns),
          ],
        },
      }),
      filters
    ),
  },
})


```